http3/response_writer.go

```diff
@@ -200,7 +203,18 @@ func (w *responseWriter) writeHeader(status int) error {
 		return err
 	}
 
+	// Handle trailer fields
+	if vals, ok := w.header["Trailer"]; ok {
+		for _, val := range vals {
+			w.declareTrailer(val)
+		}
+	}
+	w.promoteTrailers()
+
 	for k, v := range w.header {
+		if _, excluded := w.trailers[k]; excluded {
```

Comment:
Would it make sense to fold the code of `promoteTrailers` here? Then we only need to iterate over the header map once.

Reply:
Yeah, good thought. I just added this